### PR TITLE
Set cookie after login to ignore own visits

### LINKED ISSUE #100: Ignore your own visits
Hello 👋 

My analytics are biased by my own visits, making them not so reliable.

How could I block my own IP? Is there a way to set a an IP blacklist?

I found that #8 addresses this issue, but it's from 2014 and seems outdated. I can't find this setting in the current version.

Thanks

### PR DESCRIPTION
**Pull request by yehudab**
Fix #100

### CODE REVIEW

**Comment by electerious** [comment]
1) Is my assumption correct that this only works when `process.env.ACKEE_ALLOW_ORIGIN` is available? This isn't always the case as `process.env.ACKEE_ALLOW_ORIGIN` is just a fallback for those not using a reverse proxy.

2) Let's say Ackee runs on ackee.example.com. It creates a cookie for example.com because this is the matching parent domain. Would it still ignore requests from example2.com?

My idea was that Ackee creates a cookie for the current domain (e.g. ackee.example.com) and a request will contain this cookie (e.g. from example2.com) so Ackee can ignore it. I'm however not sure if that's possible, because of blocked third party cookies. I want to ensure that it works without `process.env.ACKEE_ALLOW_ORIGIN`.

**Comment by yehudab** [comment]
> 1. Is my assumption correct that this only works when `process.env.ACKEE_ALLOW_ORIGIN` is available? This isn't always the case as `process.env.ACKEE_ALLOW_ORIGIN` is just a fallback for those not using a reverse proxy.
> 
Yes, this was my plan. I wasn't aware that using `ACKEE_ALLOW_ORIGIN` is optional

> 2. Let's say Ackee runs on ackee.example.com. It creates a cookie for example.com because this is the matching parent domain. Would it still ignore requests from example2.com?
> 
In this case, only results from example.com will be ignored

> My idea was that Ackee creates a cookie for the current domain (e.g. ackee.example.com) and a request will contain this cookie (e.g. from example2.com) so Ackee can ignore it. I'm however not sure if that's possible, because of blocked third party cookies. I want to ensure that it works without `process.env.ACKEE_ALLOW_ORIGIN`.
>
Got it. I'll try a different approach and see if I cover both 1 and 2 above.

**Comment by yehudab** [comment]
Pushed another version. 
I still need to test it to make sure it's working well and to resolve the above conflicts. 
Notice that there is still a change needed in ackee-tracker part (also pushed)

**Comment by yehudab** [comment]
@electerious 
After a few more checks and fixes, I think that the PR is ready and covers the flow mentioned in your comment from yesterday.
I don't have two domains to test example.com and example2.com, but I think that it should work.

File: test/resolvers/utils.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const mongoose = require('mongoose')
+const Token = require('../../src/models/Token')
+const Domain = require('../../src/models/Domain')
+const { await } = require('signale')
+
+// Start MongoDB Instance
+const mongod = new MongoMemoryServer()
+
+// Create connection to mongoose before all tests
+exports.before = async t =>
```
Comment (electerious) [inline test/resolvers/utils.js:13]: There's a new function in src/utils/connect that can be used here.

File: src/resolvers/records.js [hunk 2]
```diff
@@ -67,6 +67,16 @@ module.exports = {
 	Mutation: {
 		createRecord: async (parent, { domainId, input }, { req }) => {
 
+			// In case of own site, return fake entry and don't track
+			if (req.cookies && req.cookies.ackee_login === "1") {
+				return {
+					success: true,
+					payload: {
+						"id": "88888888-8888-8888-8888-888888888888"
```
Comment (electerious) [inline src/resolvers/records.js:75]: Please run `npm run lint` to align the code with the rest.

File: src/resolvers/records.js [hunk 3]
```diff
@@ -100,8 +110,15 @@ module.exports = {
 			}
 
 		},
-		updateRecord: async (parent, { id }) => {
+		updateRecord: async (parent, { id },  { req }) => {
 
+			// In case of own site don't update
+			if (req.cookies && req.cookies.ackee_login === "1") {
```
Comment (electerious) [inline src/resolvers/records.js:116]: Let's create a small util for this as it's used in two places. `isLoggedIn(res)` => `true`/`false` in `src/utils/`.

**Review by electerious — COMMENTED** [review]
Great work! Just a few small things and it's ready to be merged :)

File: test/resolvers/utils.js (hunk 1 above)
Comment (yehudab) [reply test/resolvers/utils.js]: Done

File: src/resolvers/records.js (hunk 2 above)
Comment (yehudab) [reply src/resolvers/records.js]: Fixed here and in other files in this PR

File: src/resolvers/records.js (hunk 3 above)
Comment (yehudab) [reply src/resolvers/records.js]: Good idea. Done.
One small change: Parameter name will be `req`, not `res`

**Comment by electerious** [comment]
Thanks for the great work 🙌

**Comment by electerious** [comment]
There's now an ignore-own-visits branch for both Ackee and ackee-tracker. I will do some testing on my server and on Netlify. This might take a while, but I'm sure that it will be part of the next release.

The biggest outcome for me is 'MongoMemoryServer'. Haven't seen it before and it's great to add tests for all MongoDB related files. More tests are welcome (can be based on the `electerious:feature/ignore-own-visits` branch) if you want to help out with more contributions :)

**Comment by yehudab** [comment]
Happy to help with the feature and the testing.
I willl do my best to add more tests.

**Pull request merged**